### DISCUSSION TITLE
Changing value of SUAutomaticallyUpdate dynamically, and as user

### DISCUSSION THREAD

**Discussion by jeffduet** [sparkle-project/Sparkle#2361] [General]
The documentation for "SUAutomaticallyUpdate" says "Note this setting controls the default and users have the option to change this behavior." How do users change this behavior?

Also, I would like to change the value of "SUAutomaticallyUpdate" dynamically at run time, not just via the plist file, is this possibl? And if so, which API do I use? I was not able to find a API that sets this directly (I only found "automaticallyChecksForUpdates" which is a bit different).

I saw an old thread (2017) that said it can be done via the below line of code, but not sure if this still works or is recommended:

`[[NSUserDefaults standardUserDefaults] setBool:YES forKey:@"SUAutomaticallyUpdate"]`

**Comment by zorgiepoo** [comment]
> The documentation for "SUAutomaticallyUpdate" says "Note this setting controls the default and users have the option to change this behavior." How do users change this behavior?

In the software update prompt for when a new update is available there is a checkbox to allow users to opt in/out of automatically downloading updates. In Sparkle 2.4 or later, it also shows up as an option in the second-launch prompt that asks users permission to check for updates automatically.

> Also, I would like to change the value of "SUAutomaticallyUpdate" dynamically at run time, not just via the plist file, is this possibl? And if so, which API do I use? I was not able to find a API that sets this directly (I only found "automaticallyChecksForUpdates" which is a bit different).

[automaticallyDownloadsUpdates](https://sparkle-project.org/documentation/api-reference/Classes/SPUUpdater.html#/c:objc(cs)SPUUpdater(py)automaticallyDownloadsUpdates) can be used to set it dynamically. Note the documentation states: Only set this property if the user wants to change the default via a user settings option. So the use case of setting it dynamically is if you add a setting in your app that allows users to change the option. 

> I saw an old thread (2017) that said it can be done via the below line of code, but not sure if this still works or is recommended:

No it's not recommended.

**Reply by jeffduet** [reply]
Thanks much for the response. Regarding "automaticallyDownloadsUpdates", both the naming and the initial part of the documentation (i.e. "A property indicating whether or not updates can be automatically downloaded in the background.") seem to imply that this only effects downloading, not installing itself. 

So is it fair to say the documentation/naming is a bit misleading and automaticallyDownloadsUpdates does exactly what "SUAutomaticallyUpdate" does?

**Reply by zorgiepoo** [reply]
The property is indeed tied to `SUAutomaticallyUpdate` as the API documentation of the property explains. It will also try to automatically start installing the update in the background after downloading it if possible (not if authorization is required). As general note if you use any of Sparkle's methods, read the API documentation on them fully.
